### PR TITLE
Fix pypy3.8 for windows

### PR DESCRIPTION
**Pull request by mattip**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

Copying @isuruf 's comment from #59
> The windows failure is because with python 3.8, PATH is not searched for dependent DLLs and the ctypes extension cannot find libffi-8.dll.

> For CPython, our solution was to do os.add_dll_directory(sys.prefix, "Library", "bin") when the python interpreter is initialized. I'm not sure where that code should go to in PyPy though.

Thanks, the parallel place in PyPy is `pypy/interpreter/app_main.py`, which is the equivalent to `pymain_main`. I will get a patch going.

### CODE REVIEW

**Comment by mattip** [comment]
This is working: windows, osx, linux pypy3.7 and pypy3.8. Does anyone wish to review?

File: recipe/patches/0014-Add-CondaEcosystemModifyDllSearchPath.patch [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
++    import os
++
++    getenv = get_getenv()
++    debug_it = bool(getenv("CONDA_DLL_SEARCH_MODIFICATION_DEBUG"))
++    enable = getenv("CONDA_DLL_SEARCH_MODIFICATION_ENABLE")
++    if enable and enable == '0':
++        return
++    sv_executable_dirname = os.path.dirname(sys.executable)
++    condaPaths = [os.path.join(sv_executable_dirname, x) for x in [
++        r'Library\ming-w64\bin',
```
Comment (isuruf) [inline recipe/patches/0014-Add-CondaEcosystemModifyDllSearchPath.patch:41]:
```suggestion
+        r'Library\mingw-w64\bin',
```

**Comment by mattip** [comment]
@isuruf the build after fixing the typo succeeded

**Review by isuruf — APPROVED** [review]

**Comment by mattip** [comment]
Hang on, I don't see the equivalent of `CondaEcosystemModifyDllSearchPath(int add_windows_directory, int add_cwd)`, called to inject the `PATH` using `os.add_dll_directory` before calling `LoadLibraryEx`. Is that not important? It seems conda hits this code path in `ctypes.CDLL` and when importing a c-extension module.

**Comment by isuruf** [comment]
That's for the `CONDA_DLL_SEARCH_MODIFICATION_ENABLE=1` codepath, but we are not enabling that code path for pypy.
I think we'll remove that in python 3.11

**Comment by h-vetinari** [comment]
Awesome! :)

**Pull request merged**